coordination/coordination.go

```diff
@@ -11,4 +16,195 @@ type Client interface {
 	AlterNode(ctx context.Context, path string, config NodeConfig) (err error)
 	DropNode(ctx context.Context, path string) (err error)
 	DescribeNode(ctx context.Context, path string) (_ *scheme.Entry, _ *NodeConfig, err error)
+
+	// OpenSession starts a new session. This method blocks until the server session is created. The context provided is
+	// used for the lifetime of the session.
+	//
+	// To ensure resources are not leaked, one of the following actions must be performed:
+	//
+	// - cancel the provided context,
+	// - call Close on the Session,
+	// - close the Client which the session was created with,
+	// - call any method of the Session until the ErrSessionClosed is returned.
+	OpenSession(ctx context.Context, path string, opts ...options.OpenSessionOption) (Session, error)
```

Comment:
And lets to return struct/struct pointer, not interface.

Any change in interface (including add any method) - is brokable changing.
It will difficult to extend API of the Session if it will the interface.

Reply:
> 1. In the sdk and in other methods of the interface ctx is cancel token for current method call.
> 2. Timeout for the request is unusual for go: by option.
> 
> What about to use ctx in the OpenSession same way - for set cancel context/timeout of OpenSession method.
> 
> And add option for set additional context for lifetime control.
It looks a bit dangerous to me since it may cause session leaks. Though, if the whole SDK uses context that way, I think it's better to be consistent. I haven't found a way to do that for Driver or Table.Session, what are you suggestions on naming the option?